HW 9

TatyanaV24 opened the pull request:
Опубликовала решения задач по теме 9.

TatyanaV24 committed:
HW9

Решение задач по новой теме.

TatyanaV24 committed:
.

/

TatyanaV24 committed:
.

ios-review commented on HW9/HW9/ViewController.swift:
```diff
@@ -0,0 +1,25 @@
+//
+//  ViewController.swift
+//  HW9
+//
+//  Created by Татьяна Васильченко on 02.12.2020.
+// Создайте три кнопки (зёленую, синюю, жёлтую), которые находятся:
+//1. на одной горизонтали, имеют одинаковую ширину, отступ от краёв экрана 8 pt и между кнопками 4 pt (без stackviews);
+//2. на одной горизонтали, имеют одинаковую ширину, отступ слева и справа от экрана 8 pt и между кнопками 4 pt (с использованием stackviews);
+//3. на одной вертикали, занимают 50% высоты экрана, отцентрованы по вертикали, имеют отступ слева и справа от экрана 8 pt и между кнопками 4 pt (без stackviews);
+//4. на одной вертикали, занимают 50% высоты экрана, отцентрованы по вертикали, имеют отступ слева и справа от экрана 8 pt и между кнопками — 4 pt (с использованием stackviews).
```
Комментарии на скринах:

<img width="1217" alt="Screenshot 2020-12-04 at 19 26 53" src="https://user-images.githubusercontent.com/62285448/101164190-604b2180-3667-11eb-8b5a-eeaf4170502e.png">

<img width="1399" alt="Screenshot 2020-12-04 at 19 28 25" src="https://user-images.githubusercontent.com/62285448/101164247-6a6d2000-3667-11eb-8c26-22662b40ca13.png">

ios-review replied:
По третьему заданию поведение такое:
1. Нажали кнопку -> изменилась высота лейбла на высоту, равной 5 строкам.
2. Нажали кнопку -> изменилась высота лейбла на высоту, равной 1 строке.
3. Нажали кнопку -> изменилась высота лейбла на высоту, равной 5 строкам.
и т. д.

ios-review replied:
Для задания 2 дополнительно добавить два UITextField.
По изменению текста в текст филде менять значение текста в соответствующем лейбле.

Тогда можно будет проверить поведение layout во время выполнения программы.

ios-review reviewed (changes requested):
По заданию с горизонтально расположенными кнопками 👌

Над остальными можно немного поработать.

TatyanaV24 committed:
Изменения на основании комментариев

Внесла измения и разделила экран с кнопками они немного похожи расположила на разной высоте, чтобы были отличия(Item 2 и Item 3).
Для задания два добавила два UITextField, теперь есть возможность проверить верно ли решена задача.
Третья задача также решена в соответствии с описанным алгоритмом

ios-review commented on HW9/HW9/ItemFourViewController.swift:
```diff
@@ -0,0 +1,34 @@
+        let labelFrame = CGRect(x: 0, y: 0, width: 200, height: 50)
+        myLabel.frame = labelFrame
+        myLabel.numberOfLines = 1
+        myLabel.text = "Мечты всегда сбываются. Надо просто верить."
```
И действовать :)

ios-review commented on HW9/HW9/ItemFourViewController.swift:
```diff
@@ -0,0 +1,34 @@
+        } else {
+            myLabel.numberOfLines = 1
+        }
+    }
```
Есть одна багуля, если добавлять лейбл кодом и не использовать констрейнту высоты.

Попробуйте в свойство myLabel.text присвоить очень длинный текст.
А затем понажимать на кнопку.
Будет отображаться то одна, то две строки.

Хотелось бы так: то одна, то пять.

Можно сделать следующее: 
- лейбл добавить в сториборде
- добавить констрейнт высоты
- создать для констрейнта высоты @IBOutlet в коде
- связать его со сторибордом
- а затем менять значение heightConstraint.constant (значение может быть равно высоте одной строки или высоте пяти строк)

ios-review reviewed (changes requested):
Для последнего задания нужно поправить багу.

По остальным 👌

TatyanaV24 committed:
Item 5

Провела изменения в соответсвии с пунктами, которые Вы прислали.
Все работает, отоборажается 1 или 5 строк по нажатию на кнопку.
Впервые добавляю констрейнт высоты в качестве @IBOutlet, интересный прием. Скажите, а в самом XCode нет ничего подходящего, чтобы можно было использовать констрейнт высоты без его добавления? Мне казалось нечто подобное должно быть, ведь есть например numberOfLines.

ios-review reviewed (approved):
all is fine)
merge

TatyanaV24 committed:
Update
